src/sentry/snuba/metrics/datasource.py

```diff
@@ -955,6 +957,12 @@ def get_series(
 
                 results[entity][key] = {"data": snuba_result_data}
 
+    # Make sure that order-defining entities are entered first:
+    results = {
+        k: v
+        for (k, v) in sorted(results.items(), key=(lambda p: p[0] == primary_entity), reverse=True)
+    }
```

Comment:
This is a drive-by fix: I noticed that the test case below was flaky and that the endpoint did not always order groups correctly, i.e. sometimes the order of session counts was `[2, 3, 1, 1]` instead of `[3, 2, 1, 1]`.